dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/params/dimension/DimensionParamItem.java

```diff
@@ -62,17 +73,34 @@ public static List<DimensionParamItem> ofStrings(List<String> items) {
       AnalyticsQueryOperator queryOperator = getOperator(parts[0].trim());
       return singletonList(
           new DimensionParamItem(
-              queryOperator,
-              Stream.concat(Stream.of(parts[1]), items.stream().skip(1))
-                  .collect(Collectors.toList())));
+              queryOperator, Stream.concat(Stream.of(parts[1]), items.stream().skip(1)).toList()));
     } else {
       return singletonList(new DimensionParamItem(null, items));
     }
   }
 
+  /**
+   * Returns the operator for the given string.
+   *
+   * @param operator the operator string
+   * @return the operator
+   */
   private static AnalyticsQueryOperator getOperator(String operator) {
+    // If the operator starts with !, will call this method again removing the ! and then negate the
```

Comment:
I didn't understand this comment and the reason to check for "!!"...

Reply:
we want to mark operators starting with `!!` as invalid